awswrangler/athena/_write_iceberg.py

```diff
@@ -95,6 +96,12 @@ def _determine_differences(
     dtype: dict[str, str] | None,
     catalog_id: str | None,
 ) -> tuple[_SchemaChanges, list[str]]:
+    if partition_cols:
+        # Remove columns using partition transform function,
+        # as they won't be found in the DataFrame or the Glue catalog.
+        pattern = r"[A-Za-z]+\(.+\)"
```

Comment:
There are only 6 methods supported by Iceberg in Athena (listed [here](https://docs.aws.amazon.com/athena/latest/ug/querying-iceberg-creating-tables.html#querying-iceberg-creating-tables-query-editor)).

I'll make a note of it in the comments. However, I'm not sure if they'll start to support a new one that may include underscore or a number. To be honest, I don't see any downside in including them just in case, since the main thing to catch are just the parenthesis.